### ISSUE TITLE
Using wicked_pdf to generate PDF is unable to display the chart

### ISSUE THREAD

**Issue by RanX12** [mileszs/wicked_pdf#966]
This should not be an issue, it is my question, please advise, thank you!
I used in my rails project:
```ruby
gem'wicked_pdf'
gem'wkhtmltopdf-binary'
```
Their versions are:
```ruby
wicked_pdf (2.1.0)
wkhtmltopdf-binary (0.12.6.5)
```
The text content can be displayed, but I added ```echarts``` in the html, and generated a chart, and then used
```ruby
render pdf:'show', show_as_html: true, :layout =>'pdf.html.slim'
```
The chart cannot be displayed, and the downloaded PDF does not contain the chart I need. I have tried to use:
```ruby
render pdf:'show', show_as_html: true, :layout =>'pdf.html.slim', javascript_delay: 3000, disable_smart_shrinking: false
```
And set ```echarts``` in js
```js
options ={
     enableMouseTracking: false,
     shadow: false,
     animation: false,
}
```
I can’t display the chart I want in PDF, but it’s ok on the web

**Comment by RanX12** [comment]
My JS doesn't seem to be working, I used wicked_pdf_javascript_pack_tag

**Comment by unixmonkey** [comment]
Many charting libraries depend on features of HTML5 and modern JS, so it's entirely possible this chart library may not work with `wkhtmltopdf` (and by extension, WickedPDF), because the browser engine used is equivalent to something like Chrome 10. You may be able to add some Polyfill JS files to make it work.

**Comment by RanX12** [comment]
> Many charting libraries depend on features of HTML5 and modern JS, so it's entirely possible this chart library may not work with `wkhtmltopdf` (and by extension, WickedPDF), because the browser engine used is equivalent to something like Chrome 10. You may be able to add some Polyfill JS files to make it work.

thank you for your reply.
I will try to add some Polyfill JS files.

I added the following code, but js seems to be executed, but the PDF generation will always be stuck. When I end the server, I will see the PDF stuck at 50%
![image](https://user-images.githubusercontent.com/46048444/112084215-5e2fb300-8bc3-11eb-883a-67547a5a41c7.png)
![image](https://user-images.githubusercontent.com/46048444/112084379-a353e500-8bc3-11eb-89d8-fd9ea705bce9.png)
It seems that wicked_pdf does not support webpack?

**Comment by RanX12** [comment]
> Many charting libraries depend on features of HTML5 and modern JS, so it's entirely possible this chart library may not work with `wkhtmltopdf` (and by extension, WickedPDF), because the browser engine used is equivalent to something like Chrome 10. You may be able to add some Polyfill JS files to make it work.

OMG, My ruby version is: ruby 2.7.2p137 (2020-10-01 revision 5445e04352) Will it affect the use of wicked_pdf? I read the README.md mentioned: ```Wicked PDF has been verified to work on Ruby versions 2.2 through 2.6; Rails 4 through 6.1```

**Comment by unixmonkey** [comment]
> OMG, My ruby version is: ruby 2.7.2p137 (2020-10-01 revision 5445e04352) Will it affect the use of wicked_pdf? I read the README.md mentioned: `Wicked PDF has been verified to work on Ruby versions 2.2 through 2.6; Rails 4 through 6.1`

No, that just means it hasn't been extensively tested (by me) on Ruby 2.7 yet. I'm fairly sure it will work though. Please let me know if it does not, especially for simpler use cases not involving `echarts` JS.

**Comment by RanX12** [comment]
> > OMG, My ruby version is: ruby 2.7.2p137 (2020-10-01 revision 5445e04352) Will it affect the use of wicked_pdf? I read the README.md mentioned: `Wicked PDF has been verified to work on Ruby versions 2.2 through 2.6; Rails 4 through 6.1`
> 
> No, that just means it hasn't been extensively tested (by me) on Ruby 2.7 yet. I'm fairly sure it will work though. Please let me know if it does not, especially for simpler use cases not involving `echarts` JS.

It is working temporarily. I found that the reason why I am stuck at 50% when exporting PDF is because my js is too big. When I only do:
```js
import'jquery'
import echarts from'echarts/lib/echarts'
import'echarts/lib/chart/radar'
```
It is possible to export PDF, but at this time the chart generated by echarts will be deformed (compressed) and the width is not enough, as shown in the figure:
![image](https://user-images.githubusercontent.com/46048444/112570930-ead7ac80-8e21-11eb-8fa1-4549e15036d1.png)

But it is normal on the web page or in debug mode:
![image](https://user-images.githubusercontent.com/46048444/112570967-0347c700-8e22-11eb-88eb-ce1d91c457d6.png)

**Comment by unixmonkey** [comment]
You might want to experiment with using different settings for `dpi` and `zoom`.
https://github.com/mileszs/wicked_pdf#gotchas

By the way, if your JS code is trying to request any resources from the same server, you could hit a deadlock on a single webserver thread, which you get with the default `bin/rails server`. That could explain you getting stuck at 50% and hanging for a very long time. If that happens, configure your server to run 2 or more web processes.
